### DISCUSSION TITLE
How to define the fraction when using add_atom(atom, fraction)?

### DISCUSSION THREAD

**Discussion by Hegel991** [dschick/udkm1Dsim#144] [Q&A] [answered]
Hi @dschick, 
Sorry to disturb you again. I would like to know if my understanding is correct. As shown by the Section 5.3 Examples in udkm1Dsim Documentation Release 1.5.6, the Strontium Ruthenate (SRO) can be built by the script as follows,

SRO = ud.UnitCell('SRO', 'Strontium Ruthenate', c_SRO, **prop_SRO)
SRO.add_atom(O, 0)
SRO.add_atom(Sr, 0)
SRO.add_atom(O, 0.5)
SRO.add_atom(O, 0.5)
SRO.add_atom(Ru, 0.5)

I would like to know how to define the fraction (i.e., 0, 0, 0.5, 0.5, 0.5). I think that the fraction comes from the fractional coordinates of Strontium Ruthenate. The vasp file for Strontium Ruthenate obtained from the Vesta software includes the fractional coordinates. It is shown as follows,
![Position](https://github.com/dschick/udkm1Dsim/assets/79781683/0d3b5fe3-c75c-4d1a-b151-6806d8f3d0b2)
![Strontium Ruthenate](https://github.com/dschick/udkm1Dsim/assets/79781683/171cf5ab-a522-410b-8e84-65c0b0c6df21)

I am looking forward to your reply.
Best regards!

**Comment by dschick** [comment, marked as answer]
Hi @Hegel991 ,

I am happy to answer your question.
But basically, you already answered it yourself. So the fractional coordinates from the `vasp` file are correct and since only the `c`-axis matters in 1D you can omit the `a`- and `b`-axis coordinates.

Honestly, we are mostly doing the atomic positions by hand, which is quite simple for perovskites but becomes very complex for larger unit cells.  There was always the wish to have a `cif` file import or alike for simplifying this task.

When doing the atomic position by hand, you just need to count the first layer of atoms fully to the unit cell and omit the last layer. So there can be no half or quarter atoms in this model.

I am happy to help you with your real-world example.

Best

Daniel

**Reply by Hegel991** [reply]
Hi @dschick, 

Very glad to hear your answers. Thanks for your timely reply. This solved my doubts. 

Best!
